libs/platform/src/lib/components/approval-flow/approval-flow.component.ts

```diff
@@ -0,0 +1,362 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Input,
+    OnInit,
+    Output,
+    QueryList,
+    TemplateRef,
+    ViewChild,
+    ViewChildren,
+    EventEmitter,
+    OnDestroy
+} from '@angular/core';
+import { DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW, UP_ARROW } from '@angular/cdk/keycodes';
+
+import { DialogService, KeyUtil, MessageToastService } from '@fundamental-ngx/core';
+import { Subscription } from 'rxjs';
+
+import { ApprovalDataSource, ApprovalNode, ApprovalProcess, User } from './interfaces';
+import { ApprovalFlowUserDetailsComponent } from './approval-flow-user-details/approval-flow-user-details.component';
+import { ApprovalFlowNodeComponent } from './approval-flow-node/approval-flow-node.component';
+
+export type ApprovalGraphNode = ApprovalNode & { blank?: true };
+
+interface ApprovalGraphColumn {
+    nodes: ApprovalGraphNode[];
+    index?: number;
+    isPartial?: boolean;
+    allNodesApproved?: boolean
+}
+
+type ApprovalFlowGraph = ApprovalGraphColumn[];
+
+@Component({
+    selector: 'fdp-approval-flow',
+    templateUrl: './approval-flow.component.html',
+    styleUrls: ['./approval-flow.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ApprovalFlowComponent implements OnInit, OnDestroy {
+    /** Title which is displayed in the header of the Approval Flow component. */
+    @Input() title = 'Approval  process';
+
+    /** Data source for the Approval Flow component. */
+    @Input() dataSource: ApprovalDataSource;
+
+    /** A reference to the user details template */
+    @Input() userDetailsTemplate: TemplateRef<any>;
+
+    /** Event emitted on approval flow node click. */
+    @Output() nodeClick = new EventEmitter<ApprovalNode>();
+
+    /** @hidden */
+    @ViewChild('graphEl') graphEl: ElementRef;
+
+    /** @hidden */
+    @ViewChildren(ApprovalFlowNodeComponent) nodeComponents: QueryList<ApprovalFlowNodeComponent>;
+
+    /** @hidden */
+    _approvalProcess: ApprovalProcess;
+
+    /** @hidden */
+    _graph: ApprovalFlowGraph;
+
+    /** @hidden */
+    _isCarousel = false;
+
+    /** @hidden */
+    _carouselScrollX = 0;
+
+    /** @hidden */
+    _carouselStep = 0;
+
+    /** @hidden */
+    _maxCarouselStep = 0;
+
+    /** @hidden */
+    _nodeParentsMap: { [key: string]: ApprovalGraphNode } = {};
+
+    private dataSourceSub: Subscription;
+
+    /** @hidden */
+    constructor(
+        private _dialogService: DialogService,
+        private _messageToastService: MessageToastService,
+        private _cdr: ChangeDetectorRef
+    ) {
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.dataSourceSub = this.dataSource.fetch().subscribe(approvalProcess => {
```

Comment:
let's check if `this.dataSource` is presented to avoid an error in case when it's not defined.